### PR TITLE
Fix pltest on OSX

### PR DESCRIPTION
**Pull request by babsingh**
**1. Disable `j9sysinfo_test_get_levels_and_types` on OSX**

`j9sysinfo_test_get_levels_and_types` tests `j9sysinfo_get_cache_info` for
retrieving details about the cache types and levels. On OSX,
`j9sysinfo_get_cache_info` doesn't retrieve information about the cache
types and levels. So, `j9sysinfo_test_get_levels_and_types` is disabled on
OSX.

**2. Fix `j9sysinfo_test_get_cwd3` on OSX**

On OSX, "/tmp" is a symlink for "/private/tmp". On OSX,
`j9sysinfo_get_cwd` will return the full path, "/private/tmp". In
`j9sysinfo_test_get_cwd3`, "/private/tmp" should be used instead of "/tmp"
for testing `j9sysinfo_get_cwd` on OSX. `j9sysinfo_test_get_cwd3` has been
updated to account for this change.

**3. Disable j9process_testNewProcessGroupByTriggeringSignal on OSX**

`j9process_testNewProcessGroupByTriggeringSignal` sends a `SIGQUIT` signal
to the process group. When running this test on OSX, `make`, `tee` and other
test framework processes receive a `SIGQUIT` signal. This causes the test
framework to crash. Due to the incompatibility with the test framework,
this test is disabled on OSX.

**Issue:** #4520

**4. Add a new test target for the `pltest` suite on OSX**

All of the test issues in #3685 have been resolved. The `j9sock` API is
unimplemented on OSX. So, a new test target has been introduced to run
`pltest` on OSX while excluding the tests for the `j9sock` API. 

Add `<impls>...</impls>` for `cmdLineTester_pltest` and
`cmdLineTester_pltest_osx`.

**Closes:** #3685

**Note:** #4503 has been opened to implement the `j9sock` API for OSX.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by pdbain** [comment]
@babsingh 
In re your comment
> Disable j9sysinfo_test_get_levels_and_types on OSX

I recently added omrcpu_get_cache_line_size for MacOS (https://github.com/eclipse/omr/pull/3273).
Do we want similar changes for cache levels and types?

**Comment by babsingh** [comment]
> Do we want similar changes for cache levels and types?

@pdbain Currently, `j9sysinfo_get_cache_info` is only used at one place in OpenJ9 for querying the cache line size. eclipse/omr#3273 `omrcpu_get_cache_line_size` allows `j9sysinfo_get_cache_info` to query the cache line size on OSX.

https://github.com/eclipse/openj9/blob/4be8b8c9752e3fe2fa178564bcea0d1222ac51d5/runtime/vm/ObjectFieldInfo.hpp#L111-L124

In OpenJ9, `j9sysinfo_get_cache_info` is not used to retrieve the cache types and levels. So, we won't need similar changes for cache levels and types on OSX.

File: test/functional/cmdLineTests/pltest/playlist.xml [hunk 1]
```diff
@@ -46,6 +45,30 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			<subset>11</subset>
 		</subsets>
 	</test>
+	<test>
+		<testCaseName>cmdLineTester_pltest_osx</testCaseName>
```
Comment (llxia) [inline test/functional/cmdLineTests/pltest/playlist.xml]: - The command for `cmdLineTester_pltest` and `cmdLineTester_pltest_osx` are the same. Why we create a new test `cmdLineTester_pltest_osx` for osx and disable it? 
Original `cmdLineTester_pltest` is excluded on osx. If #3685 is fixed and the new issue is tracked in #4520, could we just update the comment in `cmdLineTester_pltest`?

- Question related to #4520, giving the fact that `pltest` worked on other platforms and failed on osx. Does pltest send `SIGQUIT` when running on other platforms?

Comment (babsingh) [reply test/functional/cmdLineTests/pltest/playlist.xml]: fyi - @llxia 
- `cmdLineTester_pltest_osx` is different. It excludes `j9sock` tests on OSX: `-exclude:j9sock`. it is disabled on OSX due to #4520.
- Yes, `SIGQUIT` is raised on other platforms in `j9process_testNewProcessGroupByTriggeringSignal`, which is a testcase in the `pltest` suite. Refer to https://github.com/eclipse/openj9/issues/4520#issuecomment-459092010 for the hypothesis on why the test framework crashes on OSX due to `SIGQUIT` from `pltest`.
https://github.com/eclipse/openj9/blob/ab007729a4388b5ff22b36c8c4e16c21b30c27f1/runtime/tests/port/j9processTest.c#L488-L492

**Comment by babsingh** [comment]
The Travis CI job failed because of a network issue:
```
+wget -O bootjdk11.tar.gz 'https://api.adoptopenjdk.net/v2/binary/releases/openjdk11?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk&heap_size=normal'
--2019-01-30 20:37:25--  https://api.adoptopenjdk.net/v2/binary/releases/openjdk11?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk&heap_size=normal
Resolving api.adoptopenjdk.net (api.adoptopenjdk.net)... 104.17.14.123, 104.17.15.123, 104.17.12.123, ...
Connecting to api.adoptopenjdk.net (api.adoptopenjdk.net)|104.17.14.123|:443... connected.
HTTP request sent, awaiting response... 400 Bad Request
2019-01-30 20:37:25 ERROR 400: Bad Request.
```

File: test/functional/cmdLineTests/pltest/playlist.xml [hunk 2]
```diff
@@ -46,6 +45,30 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+		<levels>
+			<level>sanity</level>
+		</levels>
+		<groups>
+			<group>functional</group>
+		</groups>
+		<subsets>
+			<subset>8</subset>
+			<subset>9</subset>
+			<subset>11</subset>
```
Comment (llxia) [inline test/functional/cmdLineTests/pltest/playlist.xml]: Please use `<subset>11+</subset>`

File: test/functional/cmdLineTests/pltest/playlist.xml (hunk 1 above)
Comment (babsingh) [reply test/functional/cmdLineTests/pltest/playlist.xml]: added a comment for merging `cmdLineTester_pltest` and `cmdLineTester_pltest_osx`.

File: test/functional/cmdLineTests/pltest/playlist.xml (hunk 2 above)
Comment (babsingh) [reply test/functional/cmdLineTests/pltest/playlist.xml]: updated.

**Comment by babsingh** [comment]
@pshipton @llxia requesting review for these changes.

**Comment by pshipton** [comment]
lgtm

File: test/functional/cmdLineTests/pltest/playlist.xml [hunk 3]
```diff
@@ -46,6 +45,29 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+			<level>sanity</level>
+		</levels>
+		<groups>
+			<group>functional</group>
+		</groups>
+		<subsets>
+			<subset>8</subset>
+			<subset>9</subset>
+			<subset>11+</subset>
+		</subsets>
```
Comment (llxia) [inline test/functional/cmdLineTests/pltest/playlist.xml]: Could you please add following in both `cmdLineTester_pltest_osx` and `cmdLineTester_pltest`?
```
<impls>
	<impl>openj9</impl>
	<impl>ibm</impl>
</impls>
```

Comment (babsingh) [reply test/functional/cmdLineTests/pltest/playlist.xml]: done.

**Comment by pshipton** [comment]
jenkins test sanity osx jdk8

**Pull request merged**